src/Symfony/Component/Runtime/SymfonyRuntime.php

```diff
@@ -0,0 +1,205 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Runtime;
+
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\ArgvInput;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\ConsoleOutput;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Dotenv\Dotenv;
+use Symfony\Component\ErrorHandler\Debug;
+use Symfony\Component\ErrorHandler\ErrorHandler;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+use Symfony\Component\HttpKernel\TerminableInterface;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class SymfonyRuntime implements RuntimeInterface
+{
+    private $request;
+    private $input;
+    private $output;
+    private $application;
+    private $command;
+
+    public function __construct(array $options = [])
+    {
+        if (isset($_SERVER['argv']) && class_exists(ArgvInput::class)) {
+            $this->getInput();
+        }
+
+        if ($path = $options['dotenv'] ?? true) {
+            if (!class_exists(Dotenv::class)) {
+                throw new \LogicException(sprintf('You cannot use "%s" as the Dotenv component is not installed. Try running "composer require symfony/dotenv".', __CLASS__));
+            }
+
+            (new Dotenv())->bootEnv(\is_string($path) ? $options['dotenv'] : (\dirname(__DIR__, 3).'/.env'));
+        }
+
+        if (!class_exists(ErrorHandler::class)) {
```

Comment:
Can we make here the same thing as we did with dotenv?  wrap this and next if clause with 
```
if ($options['error_handler'] ?? true) 
```

Reply:
I would prefer keeping it as non optional. This provides one consistent experience, that's desired. The default PHP error handler is too loose.

Reply:
I'm not sure how optional but enabled by default breaks consistent expirience. Disabling would be useful for ones who don't use these things in their console or front-controllers before for some reasones and want migrate to runtime

Reply:
More options mean more complexity. I'm not willing to make this optional.
Ppl that that want custom behavior can write a DegradedSymfonyRuntime that doesn't call this constructor it that's really what they want.

Reply:
Every IDE now marks not calling parent constructor as invalid :(

Reply:
And our brain is still superior to arbitrary rules coded by automates :)